core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponentSpec.scala

```diff
@@ -370,6 +372,27 @@ class EntityComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers wit
     }
   }
 
+  it should "time out when listing all entity types with their attribute names, if a timeout is specified" in withDefaultTestDatabase {
+    withWorkspaceContext(testData.workspace) { context =>
+      // insert a whole lot of entities with a lot of unique attribute names
+      val numEntities = 10000
+      val numAttrsPerEntity = 1000
+      (1 to numEntities) foreach { entityIdx =>
+        val attrs = (1 to numAttrsPerEntity) map { _ =>
+          val attrName = RandomStringUtils.randomAlphanumeric(12)
+          AttributeName.withDefaultNS(attrName) -> AttributeString(attrName)
+        }
+        val entity = Entity(s"entity$entityIdx", "unitTestType", attrs.toMap)
+        runAndWait(entityQuery.save(testData.workspace, entity))
+      }
+
+      // now attempt to calculate attr names and types, with a timeout of 1 second
+      intercept[MySQLTimeoutException] {
+        runAndWait(entityQuery.getAttrNamesAndEntityTypes(context.workspaceIdAsUUID, WorkspaceShardStates.Sharded, 1))
+      }
+    }
+  }
+
```

Comment:
Two thoughts- first, is it possible to write a query that runs forever? Maybe a full table scan is close enough that no DB will conceivably complete in a reasonable amount of time? Second, do we have a client side timeout on the request? Or is that what the Slick timeout is doing? One thought is to set the clientside timeout to be very short, to ensure the request times out for us if the DB response is too slow-

Reply:
Can you specify a 0 second timeout or does that just get interpreted as "don't set a timeout"? I'm assuming the smallest unit of time supported is a second?

Failing that, I'm inclined to abandon it. IMO 100% test coverage is a mirage anyway so if it is not possible to make this work after putting an honest effort in, then I don't think it's worth it given that it would test very little of our own code

Reply:
@MatthewBemis 0 seconds does equate to unbounded, so that doesn't help. I wonder if I could try a timeout of 0.001 though … 

@jordansafer we could certainly write a query that runs for an arbitrarily long time, using SQL `sleep` … but the trick is that I'm trying to test _this particular_ query, because that's the one I've modified to have a timeout. The Slick timeout is a client-side timeout (where Rawls is the client and MySQL is the server)